homeassistant/components/mqtt/discovery.py

```diff
@@ -155,116 +315,129 @@ def async_discovery_message_received(msg: ReceiveMessage) -> None:  # noqa: C901
 
         component, node_id, object_id = match.groups()
 
-        if component not in SUPPORTED_COMPONENTS:
-            _LOGGER.warning("Integration %s is not supported", component)
-            return
+        discovered_components: list[MqttComponentConfig] = []
+        if component == CONF_DEVICE:
+            # Process device based discovery message
+            device_discovery_payload = _parse_device_payload(
+                hass, payload, object_id, node_id
+            )
+            if not device_discovery_payload:
+                return
+            device_config: dict[str, Any] = device_discovery_payload[CONF_DEVICE]
+            origin_config: dict[str, Any] | None = device_discovery_payload.get(
+                CONF_ORIGIN
+            )
+            component_configs: dict[str, Any] = device_discovery_payload[
+                CONF_COMPONENTS
+            ]
```

Comment:
```suggestion
            device_config: dict[str, Any]
            origin_config: dict[str, Any] | None
            component_configs: dict[str, dict[str, Any]]
            device_config = device_discovery_payload[CONF_DEVICE]
            origin_config = device_discovery_payload.get(CONF_ORIGIN)
            component_configs = device_discovery_payload[CONF_COMPONENTS]
```

It looks like `component_configs` have to be `dict` inside `dict`s

Reply:
Thnx